src/lib/auth/refresh-access-token.ts

```diff
@@ -0,0 +1,51 @@
+import { JWT } from 'next-auth/jwt'
+
+// TODO: validate that the refresh token flow works as implemented
+
+/**
+ * Takes a token, and returns a new token with updated
+ * `accessToken` and `accessTokenExpires`. If an error occurs,
+ * returns the old token and an error property
+ *
+ * Adapted from: https://next-auth.js.org/tutorials/refresh-token-rotation#server-side
+ */
+export default async function refreshAccessToken(token: JWT) {
```

Comment:
This is not yet proved out, but it felt useful to leave it in as-is. I've got a follow-up ticket to finalize and validate the refresh token flow here: https://app.asana.com/0/1202097197789424/1202640177146339/f